dart/lib/src/io_client.dart

```diff
@@ -2,60 +2,29 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// A pure Dart client for Sentry.io crash reporting.
-import 'dart:convert';
-import 'dart:io';
+import 'package:meta/meta.dart';
 
+/// A pure Dart client for Sentry.io crash reporting.
 import 'package:sentry/sentry.dart';
+import 'package:sentry/src/transport/transport.dart';
 
 import 'client.dart';
+import 'protocol.dart';
 
 SentryClient createSentryClient(SentryOptions options) =>
     SentryIOClient(options);
 
 /// Logs crash reports and events to the Sentry.io service.
 class SentryIOClient extends SentryClient {
-  /// Instantiates a client using [dsn] issued to your project by Sentry.io as
-  /// the endpoint for submitting events.
-  ///
-  /// [environmentAttributes] contain event attributes that do not change over
-  /// the course of a program's lifecycle. These attributes will be added to
-  /// all events captured via this client. The following attributes often fall
-  /// under this category: [Event.serverName], [Event.release], [Event.environment].
-  ///
-  /// If [compressPayload] is `true` the outgoing HTTP payloads are compressed
-  /// using gzip. Otherwise, the payloads are sent in plain UTF8-encoded JSON
-  /// text. If not specified, the compression is enabled by default.
-  ///
-  /// If [httpClient] is provided, it is used instead of the default client to
-  /// make HTTP calls to Sentry.io. This is useful in tests.
-  factory SentryIOClient(SentryOptions options) => SentryIOClient._(options);
-
-  SentryIOClient._(SentryOptions options) : super.base(options);
-
-  @override
-  Map<String, String> buildHeaders(String authHeader) {
-    final headers = super.buildHeaders(authHeader);
-
-    // NOTE(lejard_h) overriding user agent on VM and Flutter not sure why
-    // for web it use browser user agent
-    headers['User-Agent'] = clientId;
+  /// Instantiates a client using [SentryOptions]
+  factory SentryIOClient(SentryOptions options) =>
+      SentryIOClient._(options, platform: sdkPlatform);
 
-    return headers;
-  }
+  static const sdk = Sdk(name: sdkName, version: sdkVersion);
 
-  @override
-  List<int> bodyEncoder(
-    Map<String, dynamic> data,
-    Map<String, String> headers,
-  ) {
-    // [SentryIOClient] implement gzip compression
-    // gzip compression is not available on browser
-    var body = utf8.encode(json.encode(data));
-    if (options.compressPayload) {
-      headers['Content-Encoding'] = 'gzip';
-      body = gzip.encode(body);
-    }
-    return body;
-  }
+  SentryIOClient._(SentryOptions options, {@required String platform})
+      : super.base(
+          options,
+          transport: Transport(options: options, sdk: sdk, platform: platform),
```

Comment:
https://github.com/getsentry/sentry-java/blob/5ebd49cbfd17eb66175e9247db6d27e4df84a56b/sentry/src/main/java/io/sentry/SentryClient.java#L43-L47

Reply:
we set the `transport` to the `options` and then pass the `options` along, so `transport` wont be null

Reply:
just to remember: we might be able to identify if its web or not on runtime via https://api.flutter.dev/flutter/foundation/kIsWeb-constant.html
so transport wont even need a platform param here, it could be already on sentry options ctor for example